packages/@aws-cdk/aws-iam/lib/user.ts

```diff
@@ -145,19 +145,13 @@ export class User extends Resource implements IIdentity, IUser {
       public readonly userArn: string = arn;
       public readonly assumeRoleAction: string = 'sts:AssumeRole';
       public readonly policyFragment: PrincipalPolicyFragment = new ArnPrincipal(arn).policyFragment;
-      private defaultPolicy?: Policy;
 
-      public addToPolicy(statement: PolicyStatement): boolean {
-        return this.addToPrincipalPolicy(statement).statementAdded;
+      public addToPolicy(_statement: PolicyStatement): boolean {
+        throw new Error('Cannot add imported User to policy');
       }
 
-      public addToPrincipalPolicy(statement: PolicyStatement): AddToPrincipalPolicyResult {
-        if (!this.defaultPolicy) {
-          this.defaultPolicy = new Policy(this, 'Policy');
-          this.defaultPolicy.attachToUser(this);
-        }
-        this.defaultPolicy.addStatements(statement);
-        return { statementAdded: true, policyDependable: this.defaultPolicy };
+      public addToPrincipalPolicy(_statement: PolicyStatement): AddToPrincipalPolicyResult {
+        throw new Error('Cannot add imported User to principal policy');
```

Comment:
The error message is the wrong way around.

`Cannot add policy to imported User principal` makes more sense to me.

Reply:
I'm sort of fine having this throw an error, but as discussed in #10913 this could just be made to work.

If you add an `AWS::IAM::Policy` and you give it the right `UserName`, it becomes possible to add an identity policy to an existing identity, as far as I can tell.

Why don't we just do that?

Maybe the correct fix is to figure out where the exception that was mentioned in #10913 is thrown, and make sure that error isn't thrown anymore. Because the content of `addToPrincipalPolicy` mostly seems correct, except it probably throws when `attachToUser` is called... (?)

Reply:
The error message will be common with `addToPolicy` function so maybe we can throw this message `Cannot add policy to imported User`.
If we don't want to throw an error then there are several problems here if we try to correct the functionality because its not like other functions are working. The `attachToUser` will call to `attachInlinePolicy` which will throw an error, I can try to fix that function...

Reply:
So, should I check for the correct functionality or throw an error instead??